presto-main/src/main/java/io/prestosql/server/DynamicFilterService.java

```diff
@@ -420,7 +420,12 @@ private void collectDynamicFilters(StageId stageId, Optional<Set<DynamicFilterId
     private static Set<DynamicFilterId> getDynamicFiltersProducedInPlanNode(PlanNode planNode)
     {
         if (planNode instanceof JoinNode) {
-            return ((JoinNode) planNode).getDynamicFilters().keySet();
+            JoinNode joinNode = (JoinNode) planNode;
+            if (joinNode.isCrossJoin()) {
```

Comment:
We should have support for ineq joins in `DynamicFilterService` too. If it's added in next commit, lets remove this one.

Reply:
Added `DynamicFilterService` support.